src/video/renderers/opengl3/gl3_renderer.c

```diff
@@ -223,8 +226,21 @@ static inline void video_set_blend_mode(gl3_context *ctx, object_array_blend_mod
         glColorMask(GL_TRUE, GL_TRUE, GL_TRUE, GL_TRUE);
     } else if(request_mode == MODE_ADD) {
         glColorMask(GL_FALSE, GL_FALSE, GL_FALSE, GL_TRUE);
-    } else {
+    } else if(request_mode == MODE_REMAP) {
+        glColorMask(GL_FALSE, GL_TRUE, GL_FALSE, GL_FALSE);
+    } else if(request_mode == MODE_SPRITE_SHADOW) {
+        glColorMask(GL_FALSE, GL_TRUE, GL_FALSE, GL_FALSE);
+        glEnable(GL_BLEND);
+        glBlendEquation(GL_MAX);
```

Comment:
Hmmh, I guess if there is some other remapping effect underneath, this can cause some interesting corruption ? This sort of feels like class-A footgun along with the round+index combining.

I can't really think of anything that would be there in arenas, that would also be picked by GL_MAX blend function though :/

Reply:
MODE_SPRITE_SHADOW is only used on the SPRITE_SHADOWs, so we'd need something to be drawn before and underneath the character's shadows.
GL_MAX would then pick whichever remap has a greater remap index-- since shadows use the first four remaps, the shadows would not overwrite whatever remap was already drawn into the buffer.

Reply:
Hum. Did the bullets hitting the ground in desert or lightning flashes in the power plant do any remapping ? I can't remember what other effects we had on the arenas. Also, the arena menu help box probably overlaps.

Reply:
Those are remapping effects, and will overwrite the shadows because they are drawn late (in MODE_REMAP, which does not do blending.)

Reply:
i thought the lightning flashes were a palette transformer, not a remap

Reply:
arena menu help box will be drawn after the shadows, and so what blend func the shadows use will not effect it.

Reply:
I think we are safe, then. Let's just go with this -- there seems to be enough comments in the code to be able to figure out any bugs if any show up later.